app/services/api/map_hesa_attributes/degrees/v01.rb

```diff
@@ -39,7 +39,7 @@ def subject_uuid
         end
 
         def graduation_year
-          if @params[:graduation_year].is_a?(String)
+          if valid_date?(@params[:graduation_year])
```

Comment:
I don't think it will ever return true.

if the input is `2003`

or have enough test to clarify intent

Reply:
Yeah it's unlikely I agree. But if someone submits "2003-04-12" or something like that, wouldn't we want to just get the graduation year from that instead of returning an error?

Reply:
Discussed IRL 

`def graduation_year` is rather a loose cannon
`2003-04-12` return `2003`
`2003` return `2003`
`Gibbish` return `Gibbish`